### PR TITLE
added gui controls to live-mix ico variables with available extractor…

### PR DESCRIPTION
**Pull request by chrispetrone**
… values.  Unable to make tabbed out extr values funcion

### CODE REVIEW

File: src/noise-filter.js [hunk 1]
```diff
@@ -46,31 +97,100 @@ export default class NoiseFilter {
...
-        const sharpness = ext.getAvg('perceptualSharpness')
-        const baseRoughness = logMap(sharpness, 0, 1, 0, 2)
+        var settingsToUpdate = {}
+
+        for (var o in this.mods){
+            var curo = this.mods[o]
+            if (curo.mapTo){
+                var newVal
+                if (curo.mapTo == 'loudness') newVal = ext.getAvg('loudness', loudness => loudness.total);
+                else newVal = ext.getAvg(`${curo.mapTo}`)
```
Comment (rickyfitts) [inline src/noise-filter.js]: No need for template string:
`else newVal = ext.getAvg(curo.mapTo)`

Comment (chrispetrone) [reply src/noise-filter.js]: thx will change that

File: src/noise-filter.js [hunk 2]
```diff
@@ -46,31 +97,100 @@ export default class NoiseFilter {
...
 
     update = ext => {
-        const loudness = ext.getAvg('loudness', loudness => loudness.total)
-        const strength = logMap(loudness, 0, 24, 1, 15)
 
-        const sharpness = ext.getAvg('perceptualSharpness')
-        const baseRoughness = logMap(sharpness, 0, 1, 0, 2)
+        var settingsToUpdate = {}
+
+        for (var o in this.mods){
```
Comment (chrispetrone) [inline src/noise-filter.js]: let

File: src/noise-filter.js [hunk 3]
```diff
@@ -46,31 +97,100 @@ export default class NoiseFilter {
...
+            var curo = this.mods[o]
```
Comment (chrispetrone) [inline src/noise-filter.js]: const curo (maybe define before for loop

File: src/noise-filter.js [hunk 4]
```diff
@@ -46,31 +97,100 @@ export default class NoiseFilter {
...
+            if (curo.mapTo){
+                var newVal
```
Comment (chrispetrone) [inline src/noise-filter.js]: let

File: src/noise-filter.js (hunk 2 above)
Comment (rickyfitts) [inline src/noise-filter.js]: use `let`, here and below, or `const` if applicable

**Pull request merged**